Source/WebCore/dom/ScriptElement.cpp

```diff
@@ -170,7 +176,21 @@ bool ScriptElement::prepareScript(const TextPosition& scriptStartPosition)
     if (wasParserInserted && !hasAsyncAttribute())
         m_forceAsync = true;
 
-    auto sourceText = scriptContent();
+    String sourceText;
```

Comment:
Both code paths below call scriptContent(), so I think we should structure the code like this:
```
String sourceText = scriptContent();
Ref context = *element().scriptExecutionContext();
if (context->settingsValues().trustedTypesEnabled && sourceText != m_scriptText) {
    auto trustedText = trustedTypeCompliantString(TrustedType::TrustedScript, context, childTextContent, is<HTMLScriptElement>(element()) ? "HTMLScriptElement text"_s : "SVGScriptElement text"_s);
    if (trustedText.hasException())
        return false;
    sourceText = trustedText.releaseReturnValue();
}
```